server/src/connectionHandlers/onReference.ts

```diff
@@ -0,0 +1,52 @@
+/* --------------------------------------------------------------------------------------------
```

Comment:
Just a note that references in an included files (.inc, bbclass) are not detected. I don't expect them to in this PR. You could consider it but it would be hard to implement.

Reply:
it should work, can you show me where it doesn't?

For the example below. the `binutils.inc` is required by `binutils_2.42.bb`:
It should work as long as `require`/`include`/`inherit` statements exist. The ones found in scan results are not implemented in this PR.
![image](https://github.com/yoctoproject/vscode-bitbake/assets/47988425/a2764e7b-d129-40ea-8f10-6cf6e677dd41)

Reply:
You are right! There are a few cases where they don't show up but I don't expect them to because of our "fast parser approach":
 1. go to reference from a .bbappend (no require/include/inherit explicited there)
 2. go to reference from a .inc doesn't reveal references in the .bb (the require is the other way around)

One way we could improve the general "fast parser approach" (the one that doesn't rely on recipe scan). Is to by default "include" all files related to the recipe at hand (all .bb/.inc/.bbappend with a matching name).

This is completly optional and not the topic of this PR. Just take note and merge this one ;)

Reply:
The features you listed are best to implement with the scan results IMO. Noted.